File: 1407-group-the-people-given-the-group-size-they-belong-to/1407-group-the-people-given-the-group-size-they-belong-to.py
class Solution:
    def groupThePeople(self, g: List[int]) -> List[List[int]]:
        mp = {}
        for i in range(0, len(g)):
            if g[i] not in mp:
                mp[g[i]] = []
            mp[g[i]].append(i)
        
        lis = []
        for aa, bb in mp.items():
            temp = []
            for i in bb:
                temp.append(i)
                if len(temp) == aa:
                    lis.append(temp)
                    temp = []
        
        return lis

File: 1407-group-the-people-given-the-group-size-they-belong-to/1407-group-the-people-given-the-group-size-they-belong-to.cpp
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& g) {
        map<int, vector<int>> mp;
        vector<vector<int>> ans;
        for(int i = 0; i < g.size(); i++) {
            mp[g[i]].push_back(i);
            if(mp[g[i]].size() == g[i]) {
                vector<int> temp;
                int c = g[i];
                while(c--) {
                    temp.push_back(mp[g[i]].back());
                    mp[g[i]].pop_back();
                }
                ans.push_back(temp);
            }
        }
        return ans;
    }
};